<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="robot" default="husky" doc="Robot name"/>
    <arg name="engine" default="torch" doc="Using NVIDIA Warp- or Pytorch-based differentiable physics"/>
    <arg name="gridmap_topic" default="height_map" doc="Input height map topic"/>
    <arg name="gridmap_frame" default="base_link" doc="Height map tf frame"/>
    <arg name="gridmap_layer" default="elevation" doc="Height map layer name"/>
    <arg name="device" default="cpu" doc="Device to run the differentiable physics on"/>
    <arg name="max_age" default="0.5"/>

    <!-- Diff Physics node -->
    <node name="diff_physics" pkg="monoforce" type="diff_physics" output="screen">
        <rosparam subst_value="true">
            robot: $(arg robot)
            gridmap_topic: $(arg gridmap_topic)
            gridmap_layer: $(arg gridmap_layer)
            max_age: $(arg max_age)
            allow_backward: true
            device: $(arg device)
            engine: $(arg engine)
        </rosparam>
    </node>
</launch>
